SAP/cloud-commerce-spartacus-storefront
# feat: New storage synchronization mechanism with context

New context aware state persistent mechanism introduced.

First implementation for the cart of this mechanism.

QA steps:
- carts should be saved in local storage and be still available after user refresh
- each base site have separate cart state saved in local storage
- you can switch between different base sites and on each you have different cart
- no `Cart not found` errors appears in the UI
- language and currency should not influence cart on the page

Closes #6215
